### PR TITLE
fix bayesStretch2 label

### PR DESCRIPTION
**Pull request by AnthonyLim23**
The new BayesStretch2 algorithm has a small mistake that mean two of the plots had the wrong x label and units. This PR fixes it. 

**To test:**
Load a red and res file for indirect
Run bayesStretch2 (keep the defaults), it can be a little slow
Plot the two workspaces in the OutputWorkspaceFit group
Check that the x unit and labels are correct (FWHM (eV) and beta), they should match the title.

*There is no associated issue.*

There are no release notes as this algorithm is new this release

---

#### Reviewer ####

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

#### Gatekeeper ####

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

### CODE REVIEW

**Comment by sf1919** [comment]
Should this be on `release-next`?

File: Framework/PythonInterface/test/python/plugins/algorithms/WorkflowAlgorithms/BayesStretch2Test.py [hunk 1]
```diff
@@ -121,8 +121,12 @@ def test_make_results(self):
         def slice(slice_list, x_data, x_unit, name):
             return slice_list
 
+        def set_label(ws_str, label, unit):
+            return ws_str
+
         self._alg.make_slice_ws = mock.Mock(side_effect=slice)
         self._alg.group_ws = mock.Mock(return_value="group")
+        self._alg.set_label = mocl.Mock(side_effect=set_label)
```
Comment (AnthonyLim23) [inline Framework/PythonInterface/test/python/plugins/algorithms/WorkflowAlgorithms/BayesStretch2Test.py:129]:
```suggestion
        self._alg.set_label = mock.Mock(side_effect=set_label)
```

File: Framework/PythonInterface/test/python/plugins/algorithms/WorkflowAlgorithms/BayesStretch2Test.py [hunk 2]
```diff
@@ -146,9 +150,11 @@ def slice(slice_list, x_data, x_unit, name):
             call_number=2,
             slice_list=FWHM_list,
             x_data=x_data,
-            x_unit="MomentumTransfer",
+            x_unit="FWHM",
             name="test_Stretch_FWHM",
         )
+        self.assert_mock_called_with(self._alg.make_slice_ws, N_calls=2, call_number=2, ws_str="test_Stretch_Beta", label="beta", unit="")
```
Comment (AnthonyLim23) [inline Framework/PythonInterface/test/python/plugins/algorithms/WorkflowAlgorithms/BayesStretch2Test.py:156]:
```suggestion
        self.assert_mock_called_with(self._alg.make_slice_ws, N_calls=2, call_number=2, ws_str="test_Stretch_Beta", label="beta", unit=x_unit)
```

File: Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/BayesStretch2.py [hunk 3]
```diff
@@ -277,8 +283,12 @@ def make_results(
         :param name: the name of the output worksapce
         :return group workspaces with the FWHM and beta slices
         """
-        beta = self.make_slice_ws(slice_list=beta_list, x_data=x_data, x_unit=x_unit, name=f"{name}_Stretch_Beta")
-        FWHM = self.make_slice_ws(slice_list=FWHM_list, x_data=x_data, x_unit=x_unit, name=f"{name}_Stretch_FWHM")
+        beta = self.make_slice_ws(slice_list=beta_list, x_data=x_data, x_unit="", name=f"{name}_Stretch_Beta")
+        FWHM = self.make_slice_ws(slice_list=FWHM_list, x_data=x_data, x_unit="FWHM", name=f"{name}_Stretch_FWHM")
+
+        FWHM = self.set_label(FWHM, "FWHM", "eV")
+        beta = self.set_label(beta, "beta", "")
```
Comment (AnthonyLim23) [inline Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/BayesStretch2.py:290]:
```suggestion
        beta = self.set_label(beta, "beta", x_unit)
```

File: Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/BayesStretch2.py [hunk 4]
```diff
@@ -277,8 +283,12 @@ def make_results(
         :param name: the name of the output worksapce
         :return group workspaces with the FWHM and beta slices
         """
-        beta = self.make_slice_ws(slice_list=beta_list, x_data=x_data, x_unit=x_unit, name=f"{name}_Stretch_Beta")
-        FWHM = self.make_slice_ws(slice_list=FWHM_list, x_data=x_data, x_unit=x_unit, name=f"{name}_Stretch_FWHM")
+        beta = self.make_slice_ws(slice_list=beta_list, x_data=x_data, x_unit="", name=f"{name}_Stretch_Beta")
```
Comment (AnthonyLim23) [inline Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/BayesStretch2.py:286]:
```suggestion
        beta = self.make_slice_ws(slice_list=beta_list, x_data=x_data, x_unit=x_unit, name=f"{name}_Stretch_Beta")
```

**Review by MialLewis — APPROVED** [review]
Fix works and is covered by tests.

File: Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/BayesStretch2.py [hunk 5]
```diff
@@ -246,6 +246,12 @@ def make_slice_ws(self, slice_list, x_data, x_unit, name):
             VerticalAxisValues=axis_names,
         )
 
+    def set_label(self, ws_str, label, unit):
+        ws = ADS.retrieve(ws_str)
+        axis = ws.getAxis(0)
+        axis.setUnit("Label").setLabel(label, unit)
+        return ws_str
```
Comment (thomashampson) [inline Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/BayesStretch2.py:253]: Why does this function need to return `ws_str`?

Comment (AnthonyLim23) [reply Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/BayesStretch2.py:253]: so we can add it to the ws group on line 277

Comment (thomashampson) [reply Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/BayesStretch2.py:253]: should it be returning `ws` instead of `ws_str`?

Comment (AnthonyLim23) [reply Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/BayesStretch2.py:253]: it works with a string and it makes the mocking nicer

**Pull request merged**